crypto/src/main/kotlin/web5/sdk/crypto/Crypto.kt

```diff
@@ -74,11 +74,11 @@ public object Crypto {
    * Computes a public key from the given private key, utilizing relevant [KeyGenerator].
    *
    * @param privateKey The private key used to compute the public key.
-   * @return The computed public key as a JWK object.
+   * @return The computed public key as a Jwk object.
    */
-  public fun computePublicKey(privateKey: JWK): JWK {
-    val rawCurve = privateKey.toJSONObject()["crv"]
-    val curve = rawCurve?.let { JwaCurve.parse(it.toString()) }
+  public fun computePublicKey(privateKey: Jwk): Jwk {
+    val rawCurve = privateKey.crv
+    val curve = JwaCurve.parse(rawCurve)
     val generator = getKeyGenerator(AlgorithmId.from(curve))
```

Comment:
calling out that this assumes we are only using keys that use ECC....which is fine for now but may break at some point in the future (if we need to use something like RSA or post quantum)

Reply:
🤯

Reply:
added a comment about this in ktdoc for `getKeyGenerator()` method here: https://github.com/TBD54566975/web5-kt/pull/262/commits/c5bba53b252324e975975f7c6f72edd4647f827c